src/vhost_user/message.rs

```diff
@@ -731,6 +731,28 @@ impl VhostUserMsgValidator for VhostUserInflight {
     }
 }
 
+/// Single memory region descriptor as payload for SET_LOG_BASE request.
+#[repr(C)]
+#[derive(Default, Clone)]
+pub struct VhostUserLog {
+    /// Size of the area to log dirty pages.
+    pub mmap_size: u64,
+    /// Offset of this area from the start of the supplied file descriptor.
+    pub mmap_offset: u64,
+}
+
+impl VhostUserLog {
+    /// Create a new instance.
+    pub fn new(mmap_size: u64, mmap_offset: u64) -> Self {
+        VhostUserLog {
+            mmap_size,
+            mmap_offset,
+        }
+    }
+}
+
+impl VhostUserMsgValidator for VhostUserLog {}
```

Comment:
Should we validate that "mmap_size + mmap_offset" doesn't wrap around?

Reply:
I can check  `mmap_size > 0`, but I'm not sure about the wrapping. `mmap_offset` is the offset in the file descriptor, so there's no invalid value there.

Reply:
If mmap_offset is offset into the `fd` and `mmap_offset + mmap_size` wraps around, the vhost-user server should fail to mmap the area.

Reply:
Oh you meant wrap around the u64 value, gotcha, I'll add this check :)

Reply:
Done!